### PR TITLE
Automatically determine cert-manager requirement

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now deploy the cert manager related resources by using the internal
manager. This way we can pro grammatically decide if cert-manager is
required or not and adjust the deployment accordingly.

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Automatically determine if cert-manager is required or not, for example in OpenShift deployments.
```

### CODE REVIEW

File: deploy/overlays/openshift/kustomization.yaml [hunk 1]
```diff
@@ -8,11 +8,3 @@ bases:
 
 patchesStrategicMerge:
   - deployment.yaml
```
Comment (saschagrunert) [inline deploy/overlays/openshift/kustomization.yaml:10]: The only difference between the OpenShift deployment and the standard one is that we enable selinux. @jhrozek do we wanna push this logic into the manager as well?

File: internal/pkg/manager/spod/spod_controller.go [hunk 2]
```diff
@@ -107,6 +110,7 @@ func (r *ReconcileSPOd) Healthz(*http.Request) error {
 //
 // OpenShift (This is ignored in other distros):
 // +kubebuilder:rbac:groups=security.openshift.io,namespace="security-profiles-operator",resources=securitycontextconstraints,verbs=use
+// +kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,verbs=get;list;watch
```
Comment (saschagrunert) [inline internal/pkg/manager/spod/spod_controller.go:113]: I dislike that. Do we have a better way to find out that we're running in OpenShift or not?

Comment (jhrozek) [reply internal/pkg/manager/spod/spod_controller.go:113]: In the ComplianceAsCode project, we detect OCP by looking for an instance of `apiVersion` `config.openshift.io/v1`, `Kind: ClusterOperator`, `name: openshift-apiserver`.  Looking for the instance might be an overkill, we do it to make sure that the version matches `4.y.z`, but I guess looking for `ClusterOperator` might be a bit safer than looking for `SCCs`.

File: deploy/overlays/openshift/kustomization.yaml (hunk 1 above)
Comment (jhrozek) [reply deploy/overlays/openshift/kustomization.yaml:10]: I think it would be nice, to enable SELinux by default through the manager as well.

Comment (saschagrunert) [reply deploy/overlays/openshift/kustomization.yaml:10]: So maybe if it’s not set explicitly in the config (not sure how to determine that), then we enable it automatically on OpenShift.

**Comment by saschagrunert** [comment]
There seems to be an issue with the operator removal test.

File: internal/pkg/manager/spod/spod_controller.go (hunk 2 above)
Comment (saschagrunert) [reply internal/pkg/manager/spod/spod_controller.go]: Fixed :+1:

**Comment by saschagrunert** [comment]
@jhrozek PTAL

File: internal/pkg/manager/spod/bindata/webhook.go [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+	"sigs.k8s.io/security-profiles-operator/internal/pkg/config"
+)
+
+var (
+	replicas                int32 = 3
+	defaultMode             int32 = 420
+	failurePolicy                 = admissionregv1.Fail
+	caBundle                      = []byte("Cg==")
+	bindindPath                   = "/mutate-v1-pod-binding"
```
Comment (jhrozek) [inline internal/pkg/manager/spod/bindata/webhook.go:41]: nit: s/bindind/binding/
OK to fix in another patch.

**Review by jhrozek — APPROVED** [review]
/lgtm
Looks good!
The only thing we need to fix as a follow-up is that the e2e tests still deploy cert-manager unconditionally, I'd like to make sure that the tests on openshift all pass even w/o it. But that's follow up work.

**Pull request closed without merging**